python/tests/engine/test_python.py

```diff
@@ -249,6 +249,68 @@ def test_read_hopsfs_connector(self, mocker):
         assert mock_python_engine_read_hopsfs.call_count == 1
         assert mock_python_engine_read_s3.call_count == 0
 
+    def test_read_hopsfs_connector_empty_dataframe(self, mocker):
+        # Arrange
+
+        # Setting list of empty dataframes as return value from _read_hopsfs
+        mock_python_engine_read_hopsfs = mocker.patch(
+            "hsfs.engine.python.Engine._read_hopsfs",
+            return_value=[pd.DataFrame(), pd.DataFrame()],
+        )
+        mock_python_engine_read_s3 = mocker.patch("hsfs.engine.python.Engine._read_s3")
+
+        python_engine = python.Engine()
+
+        connector = storage_connector.HopsFSConnector(
+            id=1, name="test_connector", featurestore_id=1
+        )
+
+        # Act
+        dataframe = python_engine.read(
+            storage_connector=connector,
+            data_format="csv",
+            read_options=None,
+            location=None,
+            dataframe_type="default",
+        )
+
+        # Assert
+        assert mock_python_engine_read_hopsfs.call_count == 1
+        assert mock_python_engine_read_s3.call_count == 0
```

Comment:
Why do we need to check that we don't read from S3?

Reply:
Not needed removed.